framework/doc/content/source/systems/NonlinearSystem.md

```diff
@@ -365,3 +365,66 @@ which there are quite a few nonlinear iterations per timestep, for which the
 cost of an additional Jacobian evaluation during the final residual evaluation
 is amortized. Also, simulations in which material property calculations are very
 expensive may be good candidates for computing the residual and Jacobian together.
+
+## Reusing preconditioners id=reuse_preconditioners
+
+The simple version of GMRES and other iterative methods converge only very
+slowly.  To improve convergence, PETSc and other iterative solver packages
+apply a [preconditioner](https://en.wikipedia.org/wiki/Preconditioner) to the
+system of equations/sparse matrix before applying the iterative solver.
+
+A great number of preconditioners exist, but 
+[multigrid](https://en.wikipedia.org/wiki/Multigrid_method)
+methods are often among the best choices.  
+The [HYPRE](application_development/hypre.md optional=true) package, 
+specifically the 
+BoomerAMG preconditioner, is often a good choice for a preconditioner to
+condition the system of equations resulting from the MOOSE simulation.
+
+A direct factorization of the sparse system of equations is a *very*
+good preconditioner.  An iterative method using the factorized matrix
+as a preconditioner will typically converge to machine precision in a single
+iteration.  However, as noted above, factorizing the sparse system of equations
+for a large simulation is numerically expensive.
+
+One option is to reuse a preconditioner for many iterations of the 
+linear iterative solver.  The preconditioner can be carried over
+through nonlinear iterations and even across load steps.  MOOSE
+allows the user to do this with the `reuse_preconditioner` flag.
+Setting
+
+```
+  reuse_preconditioner = true
+  reuse_preconditioner_max_its = 20
+```
+
+in the `[Executioner]` block will reuse the same preconditioner until
+the number of linear iterations required to solve the linearized system of
+equations exceeds 20.  At that time the solution algorithm will reform
+the preconditioner and repeat the process.
```

Comment:
```suggestion
the preconditioner and repeat the process.
```
make it clear whether it:
- in the current nonlinear iteration, accepts the result for this iteration, makes a new PC then does the next NL iteration
or
- within the nonlinear iteration, makes a new PC then tries again to reach convergence

Reply:
It will keep going until it either solves the current system or reaches `l_max_its`.  It will then form a new preconditioner in the next nonlinear iteration.

Updated the docs.